src/tests.rs

```diff
@@ -0,0 +1,136 @@
+use super::processor;
+use crate::cache::memory::MemCache;
+use crate::cache::Cache;
+use crate::http_workers::HttpWorker;
+use crate::identity;
+use crate::identity::Identity;
+use crate::proxy::ProxyWorker;
+use crate::redis;
+use crate::storage::RedisStorage;
+
+use crate::twin::{SubstrateTwinDB, Twin};
+
+use crate::anyhow::{Context, Result};
+use crate::http_api::HttpApi;
+use crate::twin::TwinDB;
+
+async fn start_rmb(
+    db: SubstrateTwinDB<MemCache<Twin>>,
```

Comment:
Why do you pass concrete types and not traits. Your function does not need to know about concrete implementations. Instead it need to accept any object that implements a certain trait. 

For example a
```rust
async fn start_rmb<D: TwinDB>(db: D)
```